### PR TITLE
Extract kicker impl

### PR DESCRIPTION
**Pull request by pope-h**
## Description   
Implemented the extract_kicker function and all accompanying tie-breaker key generators to fully support poker hand comparison and kicker extraction across every HandRank. Added helper utilities for sorting, combination generation, and value counting to enable correct tie-breaking for HIGH_CARD, ONE_PAIR, TWO_PAIR, THREE_OF_A_KIND, FULL_HOUSE, FOUR_OF_A_KIND, FLUSH, and handled STRAIGHT-type ranks as automatic ties. Also cleaned up documentation comments and ensured all new code is covered by unit tests.    

## Changes Made   - [ ] 
- Added extract_kicker implementation covering all HandRank variants

- Created get_high_card_key, get_one_pair_key, get_two_pair_key, get_three_of_a_kind_key, get_full_house_key, get_four_of_a_kind_key for rank-specific tie-breaker keys

- Implemented compare_arrays to lexicographically compare key arrays

- Added sort_cards_by_poker_value, bubble_sort (for (u16,u16,u8)), bubble_sort_u16, and bubble_sort_u8 for descending ordering

- Added generate_combinations utility for future hand-generation needs

- Updated evaluate_cards to delegate to the new kicker logic and maintain existing rank-determination behavior

- Wrote extensive unit tests covering all edge cases in tests module 

## How to Test  
Navigate to the contract folder and run `sozo test`   
 
## Checklist  
- [X] My code follows the project's coding style.  
- [X] I have tested these changes locally.   
- [X] Documentation has been updated where necessary.

### CODE REVIEW

File: poker-texas-hold-em/contract/src/models/hand.cairo [hunk 1]
```diff
@@ -96,3 +97,311 @@ impl U16HandRank of Into<u16, HandRank> {
...
+        // both hands identical → tie, no kicker
+        let cards = array![c(14, 0), c(10, 1), c(8, 2), c(4, 3), c(2, 0)];
+        let h1 = mk_hand(player1, cards.clone());
+        let h2 = mk_hand(player2, cards);
+        let (winners, kicker) = extract_kicker(
+            array![h1.clone(), h2.clone()], HandRank::HIGH_CARD.into(),
+        );
+
+        assert(winners.len() == 2, 'Both hands should tie');
+        assert(kicker.len() == 0, 'Tie means no kicker returned');
```
Comment (Birdmannn) [inline poker-texas-hold-em/contract/src/models/hand.cairo:154]: There must've been a misinformation from my side 🤦‍♂️ my fault. 
I might have a new implementation on this... the `extract_kicker` should've not just returned the winning hands, but rearrange the winning hands in a descending order. for the `rank`, we're only interested in the hand `at(0)`, but for compare hands, we'll interested in all the cards, as it'll be required for the `split`.

There are two things involved.
1. I'll merge this and adjust the `extract_kicker` by myself (or probably open a new issue for it), but when implementing `split`, you'll have to assume all kicker hands returned from `compare_hand` are ranked in descending order.
But the `split` function itself might even be a greater task... it'll require keeping track of the kicked hands and the cards or which kicked it, and calling another `extract_kicker` for re-ranking, so that the splitting of the pot is fair to all winners. But you won't be able to test this function.

2. You can adjust it knowing that what you're doing is great.

I think no. 1 would be easier 😅♠️. Keeps things going, eh?

**Review by Birdmannn — COMMENTED** [review]
That said, this is a very great job 🔥🔥 Trés bien!! 🙇🏻

File: poker-texas-hold-em/contract/src/models/hand.cairo (hunk 1 above)
Comment (pope-h) [reply poker-texas-hold-em/contract/src/models/hand.cairo:154]: Oh ok... So should i work on it or you wanna fix it. I could work on it in the evening

Comment (Birdmannn) [reply poker-texas-hold-em/contract/src/models/hand.cairo:154]: God bless you sir ☺️
In that case... #124 would suffice 😇
